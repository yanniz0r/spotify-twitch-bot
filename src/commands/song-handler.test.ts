import SongHandler from "./song-handler";
import SpotifyWebApi from "spotify-web-api-node";
import TwitchBot, { TwitchUser } from "../twitch-bot";
import TwitchConnectionAdapter from "../twitch-connection-adapter";

jest.mock("spotify-web-api-node");

describe("SongHandler", () => {
  let handler: SongHandler;
  let spotify: SpotifyWebApi;
  let bot: TwitchBot;
  let adapter: TwitchConnectionAdapter;
  let user: TwitchUser = {
    username: "yanniz0r",
  };

  beforeEach(() => {
    adapter = {
      register: jest.fn(),
      sendMessage: jest.fn(),
    };
    bot = new TwitchBot(adapter);
    spotify = new SpotifyWebApi();
    handler = new SongHandler(spotify);
  });

  it("sends a corresponding message when a track is playing", async () => {
    spotify.getMyCurrentPlayingTrack = jest.fn().mockReturnValue({
      body: {
        item: {
          name: "The Fox (What Does the Fox Say?)",
          artists: [
            {
              name: "Ylvis",
            },
          ],
        },
      },
    });
    await handler.handle(
      bot,
      {
        channel: "#test",
        text: "!song",
        user,
      },
      []
    );
    expect(adapter.sendMessage).toMatchSnapshot();
  });

  it("sends a corresponding message when no track is playing", async () => {
    spotify.getMyCurrentPlayingTrack = jest.fn().mockReturnValue({
      body: {
        item: null
      },
    });
    await handler.handle(
      bot,
      {
        channel: "#test",
        text: "!song",
        user,
      },
      []
    );
    expect(adapter.sendMessage).toMatchSnapshot();
  });
});
