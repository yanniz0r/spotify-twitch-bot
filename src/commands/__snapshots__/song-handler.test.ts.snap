// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongHandler sends a corresponding message when a track is playing 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "#test",
      "Jetzt gerade läuft \\"The Fox (What Does the Fox Say?)\\" von \\"Ylvis\\".",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`SongHandler sends a corresponding message when no track is playing 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "#test",
      "Aktuell läuft nichts auf Spotify.",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
